tasks/gitTasks.ts

```diff
@@ -146,3 +168,113 @@ export async function createPullRequest(
         return null;
     }
 }
+
+export async function getCommitFromNugetAsync(packageInfo: NugetPackageInfo): Promise<string | null> {
+    try {
+        const packageJsonString = fs.readFileSync('./package.json').toString();
+        const packageJson = JSON.parse(packageJsonString);
+        const packageVersion = packageJson['defaults'][packageInfo.packageJsonName];
+        if (!packageVersion) {
+            logError(`Can't find ${packageInfo.packageJsonName} version in package.json`);
+            return null;
+        }
+
+        const platform = await PlatformInformation.GetCurrent();
+        const vsixPlatformInfo = platformSpecificPackages.find(
+            (p) => p.platformInfo.platform === platform.platform && p.platformInfo.architecture === platform.architecture
+        )!;
+
+        const packageName = packageInfo.getPackageName(vsixPlatformInfo);
+        console.log(`${packageName} version is ${packageVersion}`);
+
+        // Nuget package should exist under out/.nuget/ since we have run the install dependencies task.
+        // Package names are always lower case in the .nuget folder.
+        const packageDir = path.join('out', '.nuget', packageName.toLowerCase(), packageVersion);
+        const nuspecFiles = fs.readdirSync(packageDir).filter((file) => file.endsWith('.nuspec'));
+
+        if (nuspecFiles.length === 0) {
+            logError(`No .nuspec file found in ${packageDir}`);
+            return null;
+        }
+
+        if (nuspecFiles.length > 1) {
+            logError(`Multiple .nuspec files found in ${packageDir}`);
+            return null;
+        }
+
+        const nuspecFilePath = path.join(packageDir, nuspecFiles[0]);
+        const nuspecFile = fs.readFileSync(nuspecFilePath).toString();
+        const results = /commit="(.*?)"/.exec(nuspecFile);
+        if (results == null || results.length === 0) {
+            logError('Failed to find commit number from nuspec file');
+            return null;
+        }
+
+        if (results.length !== 2) {
+            logError('Unexpected regex match result from nuspec file.');
+            return null;
+        }
+
+        const commitNumber = results[1];
+        console.log(`commitNumber is ${commitNumber}`);
+        return commitNumber;
+    } catch (error) {
+        logError(`Error getting commit from NuGet package: ${error}`);
+        if (error instanceof Error && error.stack) {
+            console.log(`##[debug]${error.stack}`);
+        }
+        throw error;
+    }
+}
+
+export async function createBranchAndPR(
```

Comment:
Can we just use the create branch and PR methods that already exist in this file? There isn't a dry-run functionality but I think its not the end of the world if we do not have that.